src/phoenix/db/insertion/span_annotation.py

```diff
@@ -0,0 +1,139 @@
+from datetime import datetime
+from typing import Any, List, Mapping, NamedTuple, Tuple
+
+from sqlalchemy import Select, and_, select, tuple_
+from sqlalchemy.ext.asyncio import AsyncSession
+from typing_extensions import TypeAlias
+
+from phoenix.db import models
+from phoenix.db.helpers import dedup
+from phoenix.db.insertion.types import (
+    Insertables,
+    Postponed,
+    Precursors,
+    QueueInserter,
+    Received,
+)
+
+_Key: TypeAlias = Tuple[str, str]
+_UniqueBy: TypeAlias = Tuple[str, int]
+
+
+class SpanAnnotationQueueInserter(
+    QueueInserter[
+        Precursors.SpanAnnotation,
+        Insertables.SpanAnnotation,
+        models.SpanAnnotation,
+    ],
+    table=models.SpanAnnotation,
+    unique_by=("name", "span_rowid"),
+):
+    @staticmethod
+    async def _partition(
+        session: AsyncSession,
+        retry_allowance: int,
+        *parcels: Received[Precursors.SpanAnnotation],
```

Comment:
this actually also includes Insertables, because that's just a subclass. 

In fact the word `parcel` is intended to for `Received` (not the thing that's inside Received)